libbeat/common/docker/watcher.go

```diff
@@ -119,20 +123,24 @@ func NewWatcherWithClient(client Client, cleanupTimeout time.Duration) (*watcher
 		deleted:        make(map[string]time.Time),
 		cleanupTimeout: cleanupTimeout,
 		bus:            bus.New("docker"),
+		shortID:        storeShortID,
 	}, nil
 }
 
 // Container returns the running container with the given ID or nil if unknown
 func (w *watcher) Container(ID string) *Container {
 	w.RLock()
 	container := w.containers[ID]
-	_, ok := w.deleted[ID]
+	if container == nil {
+		return nil
```

Comment:
:warning: Unlock won't be called in this case

Reply:
😱 thanks!